server/test/lib/event/event.test.js

```diff
@@ -15,4 +15,21 @@ describe('event', () => {
     // because the event emitter will be called before
     expect(alreadySeen).to.equal(false);
   });
+
+  it('should verify that event listener is removed', async () => {
```

Comment:
I don't think this is testing the behavior, because our event emitting system is async, so even if you remove the 

> eventEmitter.removeListener('test', cb);

Line, you'll see that alreadySeen will still be false

Reply:
Yes I was not sure about this test. I just found a better way by implementing and using `listenerCount`